lib/ecto/adapters/postgres/connection.ex

```diff
@@ -1255,7 +1255,14 @@ if Code.ensure_loaded?(Postgrex) do
     end
 
     def execute_ddl({:rename, %Index{} = current_index, new_name}) do
-      [["ALTER INDEX ", quote_name(current_index.name), " RENAME TO ", quote_name(new_name)]]
+      [
+        [
+          "ALTER INDEX ",
+          quote_table(current_index.prefix, current_index.name),
```

Comment:
Maybe we should combine `quote_table` and `quote_name`?  Using `quote_table` here will produce a confusing error

```elixir
defp quote_table(name) do
  if String.contains?(name, "\"") do
    error!(nil, "bad table name #{inspect(name)}")
  end

  [?", name, ?"]
end
```

Reply:
It's also used like this in the create/drop index functions, but :+1: 

As a suggestion, how about dropping `quote_table/1` entirely and using `quote_name/1` in `quote_table/2`? As the message in `quote_name/1` also includes tables, and using `quote_table/1` for the prefix as currently done in `quote_table/2` doesn't really make sense either.

Should this be done in this PR?

Reply:
To me it looks like the only difference between `quote_table` and `quote_name` is the error message and the former can use a prefix. So I think getting rid of `quote_table` altogether probably makes sense? Unless @josevalim  is seeing a reason to keep it.

Reply:
As you prefer @greg-rychlewski!

Reply:
@maltoe Sorry to add more to the PR, would you be able to get rid of all the quote_table functions and use this instead:

```elixir
defp quote_name(nil, name), do: quote_name(name)

defp quote_name(prefix, name), do: [quote_name(prefix), ?., quote_name(name)]

defp quote_name(name) when is_atom(name) do
  quote_name(Atom.to_string(name))
end

defp quote_name(name) when is_binary(name) do
  if String.contains?(name, "\"") do
    error!(nil, "bad literal/field/index/table name #{inspect(name)} (\" is not permitted)")
  end

  [?", name, ?"]
end
```

Then we can merge =)

Reply:
@greg-rychlewski Done & rebased on `master`